plan/task/rm.go

```diff
@@ -0,0 +1,69 @@
+package task
+
+import (
+	"context"
+
+	"cuelang.org/go/cue"
+	"github.com/moby/buildkit/client/llb"
+	"go.dagger.io/dagger/compiler"
+	"go.dagger.io/dagger/plancontext"
+	"go.dagger.io/dagger/solver"
+)
+
+func init() {
+	Register("Rm", func() Task { return &rmTask{} })
+}
+
+type rmTask struct {
+}
+
+func (r *rmTask) Run(ctx context.Context, pctx *plancontext.Context, s *solver.Solver, v *compiler.Value) (*compiler.Value, error) {
+	input, err := pctx.FS.FromValue(v.Lookup("input"))
+	if err != nil {
+		return nil, err
+	}
+
+	inputState, err := input.State()
+	if err != nil {
+		return nil, err
+	}
+
+	path, err := v.Lookup("path").String()
+	if err != nil {
+		return nil, err
+	}
+
+	var rmOption struct {
+		AllowWildcard bool
+
+		// FIXME(TomChv): Not correctly supported by buildkit for now
+		// See https://github.com/dagger/dagger/issues/2408#issuecomment-1122381170
+		// AllowNotFound bool
+	}
+
+	if err := v.Decode(&rmOption); err != nil {
+		return nil, err
+	}
+
+	outputState := inputState.File(
+		llb.Rm(
+			path,
+			llb.WithAllowWildcard(rmOption.AllowWildcard),
```

Comment:
nit: just realized we don't have a test that verifies setting `AllowWildcard` to `false` actually results in an error when trying to use one. While I'm 99% sure it works, we should probably add a quick test for it too.

Reply:
Actually it's does not throw an error, it's just that the `*` is not resolved, you explained me [here](https://github.com/dagger/dagger/issues/2408#issuecomment-1121528276) that's it is for the edge case where you create a file with `*` in it

> The "allowWildcards" field is necessary to handle the extreme corner case where someone creates a file or dir that actually has the char * in its name. It's more debatable whether we in Dagger should default to allowing them or not, but either way the toggle should exist IMO.

I'll add a test right

Reply:
Updated in https://github.com/dagger/dagger/pull/2422/commits/d4e9847044a4316829440e491b8103546b7fcc16